pkg/apis/stream/v1alpha1/stream_validation.go

```diff
@@ -31,16 +31,20 @@ func (s *Stream) Validate(ctx context.Context) *apis.FieldError {
 	return errs
 }
 
-func (ss *StreamSpec) Validate(ctx context.Context) *apis.FieldError {
-	if equality.Semantic.DeepEqual(ss, &StreamSpec{}) {
+func (s *StreamSpec) Validate(ctx context.Context) *apis.FieldError {
+	if equality.Semantic.DeepEqual(s, &StreamSpec{}) {
 		return apis.ErrMissingField(apis.CurrentField)
 	}
 
 	errs := &apis.FieldError{}
 
-	if ss.Provider == "" {
+	if s.Provider == "" {
 		errs = errs.Also(apis.ErrMissingField("provider"))
 	}
 
+	if s.ContentType == "" {
```

Comment:
is the content type actually required?

Reply:
yes, otherwise the invoker (which gets the start frame with the output content types computed by the processor) won't be able to find appropriate converters.
I don't think there can be a sensible default here.

Reply:
I _think_ we could try to leave this blank and have the invokers support the lack of content type.
`application/octet-stream` is also a good, equivalent, default

Reply:
done now